src/main/jbake/content/blog/2016/08/31/spark-metric-analysis.adoc

```diff
@@ -0,0 +1,141 @@
+= Metric Data Analysis using the Apache Spark
+Jiri Kremser
+2016-8-31
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, spark, metrics, cassandra
+
+The goal of this blog post is to show how to easily connect link:https://spark.apache.org/[Apache Spark] to the
+Cassandra with the metric data and do some data transformations/analysis.
+
+ifndef::env-github[]
+image::/img/blog/2016/2016-08-31-spark-logo.png[Spark logo]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/2016-08-31-spark-logo.png[Spark logo]
+endif::[]
+
+== Initialization
+First we need to create a spark context with the `spark.cassandra.connection.host` property correctly set.
+[source,scala]
+----
+val conf = new SparkConf()
+  .setAppName("HelloHawkular")
+  .setMaster("local")
+  .set("spark.cassandra.connection.host", "127.0.0.1")
+val sc = new SparkContext(conf)
+----
+
+It is very simple, the only assumption here is that the `spark-cassandra-connector` jar is on the class path.
+
+NOTE: Apache Spark is completely written in Scala so also the examples are written in Scala.
+Although, it provides also Java and Python APIs, using the language it has been written in is the best choice,
+because one can track down the calls in the (native) source code.
+
+== Reading the data from Cassandra
+
+Let's create an RDD from Cassandra table called "data" in "hawkular_metrics" keyspace. RDD stands for Resilient
+Distributed Datasets and has link:https://spark.apache.org/docs/latest/api/scala/#org.apache.spark.rdd.RDD[plenty] of
```

Comment:
"Let's create a "Resilient Distributed Dataset" (RDD) from Cassandra table called "data" in "hawkular_metrics" keyspace. Resilient Distributed Datasets have link:https://spark.apache.org/docs/latest/api/scala/#org.apache.spark.rdd.RDD[plenty] of useful operations defined."

Then you can use RDD, but the first occurence should not be the acronym.